### ISSUE TITLE
Problem setting up slack token

### ISSUE THREAD

**Issue by matheuscosta-ezops** [slackapi/hubot-slack#638]
### Description

While following the [the instructions here](https://slack.dev/hubot-slack/) I have run into the following issue when running the `HUBOT_SLACK_TOKEN=xoxb-YOUR-TOKEN-HERE ./bin/hubot --adapter slack` command.

I don't know if it is a bug or something that I'm missing but I was unable to connect the hubot with my slack.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [ ] enhancement (feature request)
- [x] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [X] I've read and understood the [Contributing guidelines](https://github.com/slackapi/hubot-slack/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [X] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

hubot-slack version: unkown 

node version: v17.3.1

OS version(s): ubuntu-focal-20.04-amd64 (ami-04505e74c0741db8d in AWS)

#### Steps to reproduce:

1. Connect to the AWS unbuntu instance
2. sudo apt-get update
3. sudo apt-get upgrade
4. sudo apt-get install python-software-properties
5. curl -fsSL https://deb.nodesource.com/setup_17.x | sudo -E bash -
6. sudo apt-get install -y nodejs
7. node –version
8. npm install --global npm@latest.
9. npm install -g yo
10. npm install -g yo generator-hubot
11. mkdir my-awesome-hubot && cd my-awesome-hubot
12. yo hubot --adapter=slack (Configure our hubot)
13. Took the slack token on our Victorops dashboard
14. HUBOT_SLACK_TOKEN=[REDACTED] ./bin/hubot --adapter slack

#### Expected result:

Hubot configured with our slack

#### Actual result:

```
HUBOT_SLACK_TOKEN=[REDACTED] ./bin/hubot --adapter slack
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'ezbot@0.0.0',
npm WARN EBADENGINE   required: { node: '0.10.x' },
npm WARN EBADENGINE   current: { node: 'v17.3.1', npm: '8.3.0' }
npm WARN EBADENGINE }
up to date, audited 162 packages in 833ms
3 packages are looking for funding
  run `npm fund` for details
2 low severity vulnerabilities
To address all issues (including breaking changes), run:
  npm audit fix --force
Run `npm audit` for details.
[Wed Jan 12 2022 14:08:46 GMT+0000 (Coordinated Universal Time)] INFO hubot-slack adapter v4.9.0
warn: SlackDataStore is deprecated and will be removed in the next major version. See project documentation for a migration guide.
warn: SlackDataStore is deprecated and will be removed in the next major version. See project documentation for a migration guide.
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 1
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 2
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 3
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 4
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 5
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 6
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 7
error: Response not OK:  method_deprecated
info: Unable to rtm.start, attempting reconnect
warn: Reconnecting, on attempt 8
info: Rate limited, will retry 60 seconds
```

**Comment by seratch** [comment]
Hi @matheuscosta-ezops, thanks for writing in!

The reason that you've encountered this is that the `rtm.start` API, which is used under the hood, is no longer available for newly created apps: https://api.slack.com/changelog/2021-10-rtm-start-to-stop

To deal with this issue at this moment is to set an env variable `HUBOT_SLACK_RTM_CLIENT_OPTS='{"useRtmConnect": "true"}'` when starting your Hubot app process. See also https://github.com/slackapi/hubot-slack/issues/632

Also, we are planning to release a new version fixing this issue shortly. Once a new version (probably v4.10) is released and you upgrade to the version, you can remove the env variable from your app start script.

I hope this helps.

**Comment by seratch** [comment]
We just released version 4.10. Try the latest version out!

**Comment by matheuscosta-ezops** [comment]
> We just released version 4.10. Try the latest version out!

Hello! Thank you so much for the quick response. We are very new to linux. How do we go about updating it to the latest version. We thought it already updated it using `npm install -g yo generator-hubot`.

In any case we ran 'sudo npm update -g'

We tried with a brand new machine and received the following text after running `~/my-awesome-hubot$ yo hubot --adapter=slack`



```
ubuntu@ip-[REDACTED]:~/my-awesome-hubot$ yo hubot --adapter=slack
(node:10927) [DEP0128] DeprecationWarning: Invalid 'main' field in '/usr/lib/node_modules/generator-hubot/node_modules/isbinaryfile/package.json' of './lib/panino.js'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:10927) [DEP0128] DeprecationWarning: Invalid 'main' field in '/usr/lib/node_modules/generator-hubot/node_modules/isbinaryfile/package.json' of './lib/panino.js'. Please either fix that or report it to the module author
(node:10927) [DEP0128] DeprecationWarning: Invalid 'main' field in '/usr/lib/node_modules/generator-hubot/node_modules/isbinaryfile/package.json' of './lib/panino.js'. Please either fix that or report it to the module author

[HUBOT SYMBOL]

? Owner [REDACTED]
? Bot name ezbot
? Description [REDACTED]
   create bin/hubot
   create bin/hubot.cmd
   create Procfile
   create README.md
   create external-scripts.json
   create hubot-scripts.json
   create .gitignore
   create package.json
   create scripts/example.coffee
   create .editorconfig

HUBOT SYMBOL

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'ezbot@0.0.0',
npm WARN EBADENGINE   required: { node: '0.10.x' },
npm WARN EBADENGINE   current: { node: 'v17.3.1', npm: '8.3.0' }
npm WARN EBADENGINE }
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
added 161 packages, and audited 162 packages in 10s
3 packages are looking for funding
  run `npm fund` for details
2 low severity vulnerabilities
To address all issues, run:
  npm audit fix
Run `npm audit` for details.

```

After we tried running `sudo npm update -hubot-slack`. Is this how we would update just the code found here?

```

ubuntu@[REDACTED]:~/my-awesome-hubot$ sudo npm update -hubot-slack
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'ezbot@0.0.0',
npm WARN EBADENGINE   required: { node: '0.10.x' },
npm WARN EBADENGINE   current: { node: 'v17.3.1', npm: '8.3.0' }
npm WARN EBADENGINE }
up to date, audited 162 packages in 6s
3 packages are looking for funding
  run `npm fund` for details
2 low severity vulnerabilities
To address all issues (including breaking changes), run:
  npm audit fix --force
Run `npm audit` for details.
```

**Comment by seratch** [comment]
@matheuscosta-ezops 
You can upgrade to the latest by running `npm i hubot-slack@~4.10` for sure. And then, check either the `npm list` command outputs or `package-lock.json` to know which version you're using now.

Also, the `npm audit` outputs, that's a different issue. You can go with the warnings for now. We may resolve the warnings in the future.
